sites/example-project/src/components/viz/ECharts.svelte

```diff
@@ -34,7 +36,10 @@
 
 <EchartsCopyTarget {config} {height} {width} {copying}/> 
 
-<DownloadData class=download-button downloadData={() => {downloadChart = true; setTimeout(() => { downloadChart = false}, 0);}} />
+<DownloadData text="Download data" {data} class=download-button />
```

Comment:
There is probably one adjustment to make here, which is to wrap the DownloadData in an if block, checking if there is in fact a data object is passed to ECharts.

The reason for this is that it is possible to create a chart by passing a config object directly to `<ECharts/>` rather than passing a `data` object. In this case there is no data object, and so the button does nothing.

Example [here](https://evidence-development-workspace-qfbf2306l-evidence-dev.vercel.app/charts/custom-charts) which is a bit contrived, but any of the [custom charts in the docs](https://docs.evidence.dev/features/charts/custom-charts) would also have this issue